### ISSUE TITLE
Rancher/Butcher drops leash when animal dies during transport

### ISSUE THREAD

**Issue by ThomasBerends** [TangoTek/TekTopia-Community#268]
**Describe the bug**
When the rancher is transporting sheep to it's pen, and it dies to 'natural causes' on the way, the leash gets dropped on the ground.

Video: https://youtu.be/aoY7Xc2RLJA

**To Reproduce**
1. Make sure the rancher has a animal to get into the pen
2. Wait until rancher gets the animal 
3. Kill animal (or let it take damage? In my case it died without it getting killed, but I've tested it while killing it)
4. See a leash drop on the ground

**Expected behavior**
The leash gets removed as it's not an item in the ranchers/butchers inventory.

** Server/Location **
Server 1, ThomasToSpace's Village

**Comment by Magleose** [comment]
Also happens with chickens

**Comment by JohnyDL** [comment]
yep free leashes here too didn't know how it happened ... this shouldn't happen
